engine/table/src/main/java/io/deephaven/engine/table/impl/sources/regioned/RegionedColumnSourceManager.java

```diff
@@ -332,68 +332,69 @@ private void pollUpdates(final RowSetBuilderSequential addedRowSetBuilder) {
-                    final long regionFirstKey = RegionedColumnSource.getFirstRowKey(regionIndex);
-                    addedRowSet.forAllRowKeyRanges((subRegionFirstKey, subRegionLastKey) -> addedRowSetBuilder
-                            .appendRange(regionFirstKey + subRegionFirstKey, regionFirstKey + subRegionLastKey));
-                    RowSet addRowSetInTable = null;
-                    try {
-                        for (final ColumnLocationState state : columnLocationStates) {
-                            if (state.needToUpdateGrouping()) {
-                                state.updateGrouping(
-                                        addRowSetInTable == null ? addRowSetInTable = updateRowSet.shift(regionFirstKey)
-                                                : addRowSetInTable);
+                    if (log.isDebugEnabled()) {
+                        log.debug().append("LOCATION_SIZE_CHANGE:").append(location.toString())
+                                .append(",FROM:").append(rowSetAtLastUpdate.size())
+                                .append(",TO:").append(updateRowSet.size()).endl();
+                    }
+                    try (final RowSet addedRowSet = updateRowSet.minus(rowSetAtLastUpdate)) {
+                        final long regionFirstKey = RegionedColumnSource.getFirstRowKey(regionIndex);
+                        addedRowSet.forAllRowKeyRanges((subRegionFirstKey, subRegionLastKey) -> addedRowSetBuilder
+                                .appendRange(regionFirstKey + subRegionFirstKey, regionFirstKey + subRegionLastKey));
+                        RowSet addRowSetInTable = null;
+                        try {
+                            for (final ColumnLocationState state : columnLocationStates) {
+                                if (state.needToUpdateGrouping()) {
+                                    state.updateGrouping(
+                                            addRowSetInTable == null
+                                                    ? addRowSetInTable = updateRowSet.shift(regionFirstKey)
+                                                    : addRowSetInTable);
+                                }
+                            }
+                        } finally {
+                            if (addRowSetInTable != null) {
+                                addRowSetInTable.close();
                             }
-                        }
-                    } finally {
-                        if (addRowSetInTable != null) {
-                            addRowSetInTable.close();
                         }
                     }
+                } finally {
+                    rowSetAtLastUpdate.close();
+                    rowSetAtLastUpdate = updateRowSet;
                 }
-            } finally {
-                rowSetAtLastUpdate.close();
-                rowSetAtLastUpdate = updateRowSet;
+            } catch (Exception ex) {
+                poison();
+                throw ex;
             }
```

Comment:
I think this is heavy-handed and unnecessary.

Reply:
In the words of the immortal bard Sheryl Crow: If it makes you happy, it can't be that bad.